prime-router/src/test/kotlin/fhirengine/azure/FHIRTranslatorIntegrationTests.kt

```diff
@@ -0,0 +1,611 @@
+                format = MimeFormat.HL7
+            )
+        )
+        val receivers = UniversalPipelineTestUtils.createReceivers(receiverSetupData)
+        val org = UniversalPipelineTestUtils.createOrganizationWithReceivers(receivers)
+        val translator = createFHIRTranslator(azureEventService, org)
+        val reportContents = File(HL7_WITH_BIRTH_TIME).readText()
+        val receiveReport = UniversalPipelineTestUtils.createReport(
+            reportContents,
+            TaskAction.receive,
+            Event.EventAction.CONVERT,
+            azuriteContainer,
+            fileName = "originalhl7.hl7"
+        )
+        val queueMessage = generateQueueMessage(
+            receiveReport,
+            reportContents,
+            UniversalPipelineTestUtils.hl7SenderWithSendOriginal,
+            "phd.x"
+        )
+        val fhirFunctions = UniversalPipelineTestUtils.createFHIRFunctionsInstance()
+
+        // execute
+        fhirFunctions.doTranslate(queueMessage, 1, translator)
+
+        // check action table
+        UniversalPipelineTestUtils.checkActionTable(listOf(TaskAction.receive, TaskAction.translate))
+
+        // verify task and report_file tables were updated correctly in the Translate function (new task and new
+        // record file created)
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val batchTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
+                .where(Task.TASK.NEXT_ACTION.eq(TaskAction.batch))
+                .fetchOneInto(Task.TASK)
+            // verify batch queue task does not exist
+            assertThat(batchTask).isNull()
+
+            val sendTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
+                .where(Task.TASK.NEXT_ACTION.eq(TaskAction.send))
+                .fetchOneInto(Task.TASK)
+            // verify send queue task exists
+            assertThat(sendTask).isNotNull()
+
+            val sendReportFile =
+                DSL.using(txn).select(ReportFile.REPORT_FILE.asterisk())
+                    .from(ReportFile.REPORT_FILE)
+                    .where(
+                        ReportFile.REPORT_FILE.REPORT_ID
+                            .eq(sendTask!!.reportId)
+                    )
+                    .fetchOneInto(ReportFile.REPORT_FILE)
+            assertThat(sendReportFile).isNotNull()
+
+            // verify message format is HL7
+            assertThat(sendTask.bodyFormat).isEqualTo("HL7")
+
+            // verify message matches the original HL7 input
+            val translatedValue = BlobAccess.downloadBlobAsByteArray(
+                sendReportFile!!.bodyUrl,
+                UniversalPipelineTestUtils.getBlobContainerMetadata(azuriteContainer)
+            )
+            assertThat(translatedValue).isEqualTo(reportContents.toByteArray())
+        }
+    }
+
+    @Test
+    fun `successfully translate for FHIR receiver when isSendOriginal is true`() {
```

Comment:
Maybe one more test verifying the sendOriginal behavior when the original is HL7, but the receiver is configured to get FHIR?